### PR TITLE
ROSAENG-61180 | test: add tests for create operatorroles by prefix and role helpers

### PR DESCRIPTION
**Pull request by olucasfreitas**
## PR Summary
Add Ginkgo v2 unit tests for `by_prefix.go` validation and ARN helpers, expand `common_utils_test.go`, and add a new test suite for `pkg/interactive/roles/roles.go` — 19 test scenarios total.

## Detailed Description of the Issue
`cmd/create/operatorroles/by_prefix.go` (636L, largest file in scope) and `pkg/interactive/roles/roles.go` (77L) had **0% coverage**. `common_utils.go` had partial coverage missing `validateIngressOperatorPolicyOverride`. This PR adds tests for `computePolicyARN` (4 partition scenarios), `validateArgumentsOperatorRolesCreationByPrefix` (valid path), `convertCredRequestsOperatorRolesIntoV1OperatorIAMRole` (3 scenarios), `validateIngressOperatorPolicyOverride` (7 scenarios covering policy-not-found, invalid JSON, matching/different ARNs, deny effects), and `GetInstallerRoleArn` (4 scenarios).

## Related Issues and PRs
- Jira: [ROSAENG-61180](https://redhat.atlassian.net/browse/ROSAENG-61180)
- Parent: [ROSAENG-60112](https://redhat.atlassian.net/browse/ROSAENG-60112)
- Stack: **4/7** in the [ROSAENG-61180 stack](https://github.com/openshift/rosa/pull/3521)
- Related PR(s): #3521 #3522 #3523 #3525 #3526 #3527

## Type of Change
- [x] test - adds or updates tests only.

## Previous Behavior
Prefix validation, ARN computation, ingress operator policy override validation, and installer role ARN helpers had no unit test coverage.

## Behavior After This Change
No user-facing behavior change. 19 new unit tests cover helper functions and validation logic with edge cases (govcloud partitions, custom paths, invalid JSON policies, deny effects).

### Preconditions
Go toolchain matching `go.mod`, `make install-hooks` run.

### Test Steps
1. `make test` — full suite passes
2. `go test -v ./cmd/create/operatorroles/...` — all operatorroles specs pass
3. `go test -v ./pkg/interactive/roles/...` — 4 new specs pass
4. `make lint` — 0 issues

### Expected Results
All tests green, no lint issues.

## Proof of the Fix
- Logs/CLI output: operatorroles 17 new specs + roles 4 new specs all pass

## Breaking Changes
- [x] No breaking changes

## Developer Verification Checklist
- [x] Commit subject/title follows `[JIRA-TICKET] | [TYPE]: <MESSAGE>`.
- [x] PR description clearly explains both **what** changed and **why**.
- [x] Relevant Jira/GitHub issues and related PRs are linked.
- [x] `make install-hooks` has been run in this clone.
- [x] Tests were added/updated where appropriate.
- [x] I manually tested the change.
- [x] `make test` passes.
- [x] `make lint` passes.
- [x] `make rosa` passes.
- [x] Documentation or repo-local agent guidance was added/updated where appropriate.
- [x] Any risk, limitation, or follow-up work is documented.

[ROSAENG-61180]: https://redhat.atlassian.net/browse/ROSAENG-61180?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ROSAENG-60112]: https://redhat.atlassian.net/browse/ROSAENG-60112?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by amandahla** [comment]
@coderabbitai Review the test coverage for cmd/create/operatorroles/by_prefix.go in this PR.

  Focus only on behavior implemented or orchestrated by by_prefix.go. Do not count tests of lower-level AWS/OCM helpers in other packages as coverage of this file unless they invoke the by_prefix.go flow directly.

  Check whether tests cover these scenarios:

  1. CreateOperatorRoles:
     - propagates all input parameters into the prefix-based flow.

  2. HandleOperatorRoleCreationByPrefix:
     - retrieves the OIDC config and uses its issuer URL;
     - validates prefix, OIDC HTTPS URL, and installer role ARN;
     - extracts installer role name/path and rejects non-ROSA-standard installer roles;
     - gets credential requests, managed-policy state, and AWS creator;
     - validates generated operator roles against the OIDC provider;
     - allows NoSuchEntityException but rejects other validation errors.

  3. Auto mode:
     - creates non-managed policies, IAM roles, and policy attachments;
     - uses managed policy ARNs when applicable;
     - uses ForceEnsurePolicy when force-policy-creation is enabled;
     - logs correct OCM success/failure events, including throttling classification;
     - supports HCP policies and validates HasHostedCPPolicies;
     - handles HCP Shared VPC policy creation and attachments.

  4. Manual mode:
     - generates policy files and AWS CLI commands;
     - handles existing and missing policies;
     - handles managed policies;
     - handles HCP Shared VPC commands and avoids duplicate shared-VPC policy creation.

  5. Error and edge cases:
     - OIDC config, credential-request, managed-policy, creator, and HCP-policy lookup failures;
     - invalid mode;
     - invalid prefix, non-HTTPS OIDC URL, and invalid installer role ARN;
     - errors from policy/role creation and command/file generation.

  For each missing scenario, report:
  - severity: critical, warning, or info;
  - the exact function and branch not covered;
  - whether an existing test elsewhere in the repository already covers it indirectly;
  - a concise, concrete test case recommendation.

  Do not recommend tests for behavior outside cmd/create/operatorroles/by_prefix.go.

File: cmd/create/operatorroles/common_utils_test.go [hunk 1]
```diff
@@ -61,3 +63,117 @@ var _ = Describe("Create dns domain", func() {
...
+		testInstallerPfx = "my-prefix"
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		runtime = rosa.NewRuntime()
+		mockClient = aws.NewMockClient(ctrl)
+		runtime.AWSClient = mockClient
+		mockClient.EXPECT().GetCreator().Return(&aws.Creator{Partition: "aws"}, nil)
+		creator, err := runtime.AWSClient.GetCreator()
```
Comment (amandahla) [inline cmd/create/operatorroles/common_utils_test.go]: @coderabbitai is this needed? (creator)

File: cmd/create/operatorroles/common_utils_test.go [hunk 2]
```diff
@@ -61,3 +63,117 @@ var _ = Describe("Create dns domain", func() {
...
+
+	When("the policy does not exist", func() {
+		It("returns nil without further checks", func() {
+			mockClient.EXPECT().IsPolicyExists(testPolicyArn).Return(nil, fmt.Errorf("NoSuchEntity"))
+			err := validateIngressOperatorPolicyOverride(runtime, testPolicyArn, testSharedVpcArn, testInstallerPfx)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	When("the policy exists", func() {
```
Comment (amandahla) [inline cmd/create/operatorroles/common_utils_test.go:96]: Consider adding a scenario with multiple policy statements—for example, an irrelevant statement followed by an Allow / sts:AssumeRole statement with an unexpected shared VPC role ARN. This verifies that the loop does not miss a violation when it is not
  in the first statement.

@coderabbitai check if makes sense, short answer

File: pkg/interactive/roles/roles_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/interactive/confirm"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+var _ = Describe("GetInstallerRoleArn", func() {
```
Comment (amandahla) [inline pkg/interactive/roles/roles_test.go:19]: @coderabbitai 
Review the test coverage for pkg/interactive/roles/roles.go, specifically GetInstallerRoleArn.

  Please verify whether the PR includes coverage for these missing paths:

  - findRoleARNs returns an error: the function reports the error and exits with os.Exit(1);
  - no roles are found: the function reports the warning and exits with os.Exit(1);
  - multiple roles are found with --yes enabled, but none uses the default prefix: the first returned role must be selected;
  - multiple roles are found without --yes: the function must use interactive.GetOption;
  - in the interactive multiple-role path, defaultInstallerRoleArn must be passed as the default selected option;
  - one of multiple role ARNs is malformed: aws.GetResourceIdFromARN fails for that entry, it is ignored while searching for the default-prefixed role, and the flow continues.

  For every uncovered path, report the exact branch in GetInstallerRoleArn and suggest a focused test case. Do not suggest changes outside pkg/interactive/roles/roles.go and its corresponding tests.
 Prioritize straightforward, maintainable unit tests using existing repository patterns. Do not recommend tests that require complex mocking, subprocess orchestration solely to intercept os.Exit, or brittle simulation of interactive terminal input, unless
  the repository already has an established test helper for that pattern.

Comment (amandahla) [reply pkg/interactive/roles/roles_test.go:19]: @olucasfreitas could you add the missing non-interactive tests? (lines 3 and 5)

File: cmd/create/operatorroles/common_utils_test.go (hunk 2 above)
Comment (amandahla) [reply cmd/create/operatorroles/common_utils_test.go:96]: @olucasfreitas could you add this test?

File: cmd/create/operatorroles/common_utils_test.go (hunk 1 above)
Comment (amandahla) [reply cmd/create/operatorroles/common_utils_test.go]: @olucasfreitas since is not needed, could you remove it?
@coderabbitai also check other files in this PR if there is something in beforeeach that is not used.

File: cmd/create/operatorroles/by_prefix_test.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+			}
+
+			roleList, err := convertCredRequestsOperatorRolesIntoV1OperatorIAMRole(
+				credRequests, "test-prefix", creator, "",
+			)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(roleList).To(HaveLen(1))
+			Expect(roleList[0].Name()).To(Equal("cloud-credentials"))
+			Expect(roleList[0].Namespace()).To(Equal("openshift-cloud-credential-operator"))
+			Expect(roleList[0].RoleARN()).ToNot(BeEmpty())
```
[the remark this answers is not in the document]
Comment (olucasfreitas) [reply cmd/create/operatorroles/by_prefix_test.go]: Done — single- and multi-role conversion tests now assert the exact ARN (via `aws.ComputeOperatorRoleArn`), plus name and namespace for each entry.

File: cmd/create/operatorroles/common_utils_test.go (hunk 1 above)
Comment (olucasfreitas) [reply cmd/create/operatorroles/common_utils_test.go]: Removed the unused `GetCreator()` / `runtime.Creator` setup from `validateIngressOperatorPolicyOverride` tests. Also dropped the same unused pattern from `by_prefix_test` BeforeEach; `getHcpSharedVpcPolicy` tests now set `runtime.Creator` directly since that helper needs partition/account ID.

File: cmd/create/operatorroles/common_utils_test.go (hunk 2 above)
Comment (olucasfreitas) [reply cmd/create/operatorroles/common_utils_test.go:96]: Added a case with an irrelevant statement first, then an `Allow` / `sts:AssumeRole` statement with an unexpected shared VPC role ARN; the test expects the validation error.

File: pkg/interactive/roles/roles_test.go (hunk 3 above)
Comment (olucasfreitas) [reply pkg/interactive/roles/roles_test.go:19]: Added the two non-interactive cases: (1) multiple roles with `--yes` and no default-prefixed role falls back to the first ARN, and (2) a malformed ARN is skipped while the default-prefixed role is selected. Left exit/interactive branches untested per the agreed scope.
